website/project/views/contributor.py

```diff
@@ -867,19 +876,28 @@ def claim_user_form(auth, **kwargs):
 def _add_related_claimed_tag_to_user(pid, user):
     """
     Adds claimed tag to incoming users, depending on whether the resource has related source tags
-    :param pid: guid of either the node or the preprint
+    :param pid: guid of either the node or the preprint or the registration
     :param user: the claiming user
     """
     node = AbstractNode.load(pid)
     preprint = Preprint.load(pid)
+    registration = Registration.load(pid)
     osf_claimed_tag, created = Tag.all_tags.get_or_create(name=provider_claimed_tag('osf'), system=True)
-    if node:
+    if registration:
+        registration_id = registration.provider._id
```

Comment:
Could this be renamed to registration_provider_id?

Reply:
Yes, good call